### DISCUSSION TITLE
LISF Configuration and installation

### DISCUSSION THREAD

**Discussion by dipeshpandit12** [NASA-LIS/LISF#1598] [Q&A] [answered]
I have installed all required dependencies and cloned the LISF from the GitHub. I have stored it in LISF folder. Now, this is what I do ls command in the LISF folder:
![image](https://github.com/user-attachments/assets/72b75bd8-31fd-45e0-a016-b0108b64bcba)

What is my next step for installation or has it been installed successfully?
Note: I have not installed the below packages. I have only followed: 
https://nasa-lis.github.io/LISF/LISF_installation_guide/LISF_installation_guide.html

[LDT Users’ Guide](https://nasa-lis.github.io/LISF/LDT_users_guide/LDT_users_guide.html)

[LIS Users’ Guide](https://nasa-lis.github.io/LISF/LIS_users_guide/LIS_users_guide.html)

[LVT Users’ Guide](https://nasa-lis.github.io/LISF/LVT_users_guide/LVT_users_guide.html)

**Comment by dmocko** [comment]
HI @dipeshpandit12 

Thanks for your interest in LIS.  You have copied the software, but you next need to install the dependencies/packages
as listed in the Installation Guide.  You then need to compile and configure the LDT, LIS, and LVT executables separately,
following the directions in their respective Users' Guides.

**Reply by dipeshpandit12** [reply]
I am facing **./compile** problem, I have attached the error image:
======================================================================
Note:
- I have followed -disable compression in Netcdf following the step 3.1.4:
https://nasa-lis.github.io/LISF/LISF_installation_guide/LISF_installation_guide.html
=======================================================================
Path:
1. export LDT_FC=/cm/local/apps/gcc/11.2.0/bin/gfortran
2. export LDT_CC=/cm/local/apps/gcc/11.2.0/bin/gcc
3. export LD_LIBRARY_PATH=/home/rab347/LISF/netcdf-fortran-4.5.3
4. LDT_LIBESMF=/home/rab347/LISF/esmf-8.6.1/lib/libO/Linux.gfortran.64.mpiuni.default
5. export LDT_MODESMF=/home/rab347/LISF/esmf-8.6.1/mod/modO/Linux.gfortran.64.mpiuni.default/
6. export LDT_OPENJPEG=/home/rab347/LISF/openjpeg-2.5.2/build/
7. export LDT_ECCODES=/home/rab347/LISF/build/
8. export LDT_NETCDF=/home/rab347/LISF/netcdf-fortran-4.5.3
9. export LD_LIBRARY_PATH=$LDT_LIBESMF:$LDT_NETCDF:$LDT_OPENJPEG:$LDT_ECCODES:$LD_LIBRARY_PATH
====================================================================


![image](https://github.com/user-attachments/assets/5a77570f-db24-4d14-aad4-96ec9de044bc)

**Comment by jvgeiger** [comment]
Hello.  Your error message says that the linker cannot find the NetCDF library.  Please check where you installed both the  NetCDF-C and NetCDF-Fortran libraries.  The build scripts expect both libraries to be in the same location.

**Reply by dipeshpandit12** [reply]
It is still showing the error :
----------------------------------------------
1. I have installed both the NetCDF-C and NetCDF-Fortran libraries in this dir:
/home/rab347/LISF/netcdf

Note: inorder to installed both libraries in the above dir , I used the following commands:

NetCDF-C  configure step:
pwd: /home/rab347/LISF/netcdf-c-4.9.2/

./configure --prefix=/home/rab347/LISF/netcdf --disable-netcdf-4 --disable-byterange --disable-hdf5
gmake
gmake install

NetCDF-Fortran configure step:
pwd: /home/rab347/LISF/netcdf-fortran-4.5.3/

export LDFLAGS=-L/home/rab347/LISF/netcdf/lib
export CPPFLAGS=-I/home/rab347/LISF/netcdf-c-4.9.2/include/
./configure --prefix=/home/rab347/LISF/netcdf
gmake
gmake install

 --------------------------------------------------------------------------------

2. "/home/rab347/LISF/netcdf " includes the sub dir such as:
![image](https://github.com/user-attachments/assets/653501a2-67f5-4293-8db0-c1899b2ee164)

-----------------------------------------------------------------------------

3. I update the path:

export LDT_FC=/cm/local/apps/gcc/11.2.0/bin/gfortran
export LDT_CC=/cm/local/apps/gcc/11.2.0/bin/gcc
export LD_LIBRARY_PATH=home/rab347/LISF/netcdf/lib
export LDT_LIBESMF=/home/rab347/LISF/esmf-8.6.1/lib/libO/Linux.gfortran.64.mpiuni.default
export LDT_MODESMF=/home/rab347/LISF/esmf-8.6.1/mod/modO/Linux.gfortran.64.mpiuni.default/


export LDT_OPENJPEG=/home/rab347/LISF/openjpeg-2.5.2/build
export LDT_ECCODES=/home/rab347/LISF/build
export LDT_NETCDF=/home/rab347/LISF/netcdf
export LD_LIBRARY_PATH=$LDT_LIBESMF:$LDT_NETCDF:$LDT_OPENJPEG:$LDT_ECCODES:$LD_LIBRARY_PATH

-----------------------------------------------------------------------------------------------
 4. when I tried ./configure and ./compile in /home/rab347/LISF/TOPLEVELDIR/LISF/ldt, shows the below error:
 
![image](https://github.com/user-attachments/assets/9ebc3ba5-61b8-4b20-ab28-7b7e2951c2b6)

**Comment by jvgeiger** [comment]
Hello,

First, it seems that you are using OpenJPEG and ecCodes from their CMAKE build directories.   The LDT build scripts should be able to find the ecCodes libraries, but they will not find the OpenJPEG libraries.  Please go into your /home/rab347/LISF/openjpeg-2.5.2/build directory and run `ln -s bin lib`

Now, I sincerely apologize.  You are trying to compile a configuration that I have not tested in a while.  And I see that there are several other compiling issues that must be fixed.  The easiest way forward for you is:

1. Install HDF5

```
./configure --prefix=/home/rab347/LISF/hdf5 --enable-fortran
gmake
gmake install
```

2. Reinstall NetCDF-C

```
CPPFLAGS=-I/home/rab347/LISF/hdf5/include LDFLAGS=-L/home/rab347/LISF/hdf5/lib ./configure --prefix=/home/rab347/LISF/netcdf --enable-netcdf-4
gmake
gmake install
```

3. Reinstall NetCDF-Fortran (as you did before)

```
CPPFLAGS=-I/home/rab347/LISF/netcdf/include LDFLAGS=-L/home/rab347/LISF/netcdf/lib ./configure --prefix=/home/rab347/LISF/netcdf
gmake
gmake install
```

4. Then you must set

```
export LDT_HDF5=/home/rab347/LISF/hdf5
export LD_LIBRARY_PATH=$LDT_LIBESMF:$LDT_HDF5/lib:$LDT_NETCDF/lib:$LDT_OPENJPEG/lib:$LDT_ECCODES/lib:$LD_LIBRARY_PATH
```

5. Finally, rebuild LDT.  In the LDT source directory:

```
gmake -C make realclean
./configure
./compile
```

Note: At the `Use HDF5? (1-yes, 0-no, default=1):` prompt, select 1.  At the `NETCDF version (3 or 4, default=4)?:`, select 4.  And I suggest:

```
NETCDF use shuffle filter? (1-yes, 0-no, default = 1): 1
NETCDF use deflate filter? (1-yes, 0-no, default = 1): 1
NETCDF use deflate level? (1 to 9-yes, 0-no, default = 9): 1
```

**Reply by dipeshpandit12** [reply]
Is this the problem with the setting environmental variables of ESMF Library?

![image](https://github.com/user-attachments/assets/8902bead-a1e3-42a8-8e0e-3f1c10962381)

**Reply by dipeshpandit12** [reply]
I reinstalled ESMF assigning the environmental variables, and ESMF installed successfully but still, the ./compile failed in ldt :
=================================================
ESMF_DIR=/home/rab347/LISF/esmf-8.6.1
ESMF_BOPT=g
ESMF_COMM=mpich
ESMF_COMPILER=gfortran
export ESMF_F90COMPILER=/cm/local/apps/gcc/11.2.0/bin/gfortran
export ESMF_CXXCOMPILER=/cm/local/apps/gcc/11.2.0/bin/gcc
ESMF_ABI=64
ESMF_SITE=
ESMF_INSTALL_PREFIX=/home/rab347/LISF/esmf-8.6.1
=================================================
Variables declared before ./configure :

export LDT_LIBESMF=/home/rab347/LISF/esmf-8.6.1/lib/libO/Linux.gfortran.64.mpiuni.default
export LDT_MODESMF=/home/rab347/LISF/esmf-8.6.1/mod/modO/Linux.gfortran.64.mpiuni.default
export LDT_OPENJPEG=/home/rab347/LISF/openjpeg-2.5.2/build
export LDT_ECCODES=/home/rab347/LISF/build
export LDT_NETCDF=/home/rab347/LISF/netcdf
export LDT_HDF5=/home/rab347/LISF/hdf5
export LD_LIBRARY_PATH=$LDT_LIBESMF:$LDT_HDF5/lib:$LDT_NETCDF/lib:$LDT_OPENJPEG/lib:$LDT_ECCODES/lib:$LD_LIBRARY_PATH
=================================================

![image](https://github.com/user-attachments/assets/eaa287d2-a302-4e0a-8e91-3ec39432cd2b)

**Comment by jvgeiger** [comment, marked as answer]
Hello.  Please try editing your ldt/make/configure.ldt file.  Go to the line that defines LDFLAGS and add `-lrt` right before `-lz`.  Then rerun the `./compile` script.  Do *not* rerun the `./configure` script.

**Reply by dmocko** [reply]
But please be sure to run "gmake realclean" in the ldt/make directory before re-compiling.

**Reply by dipeshpandit12** [reply]
It worked, Thank you so much for your help.
